eventmetrics/parsers/timeElapsedParser.go

```diff
@@ -77,13 +77,13 @@ type TimeElapsedParser struct {
 	incomingEvent EventInfo
 	searchedEvent EventInfo
 	name          string
-	logger        log.Logger
+	logger        *zap.Logger
 	client        EventClient
 	measures      Measures
 }
 
 // NewTimeElapsedParser creates a TimeElapsedParser from a TimeElapsedConfig or returns an error if there one cannot be created.
-func NewTimeElapsedParser(config TimeElapsedConfig, client EventClient, logger log.Logger, measures Measures, currentTime TimeFunc) (*TimeElapsedParser, error) {
+func NewTimeElapsedParser(config TimeElapsedConfig, client EventClient, logger *zap.Logger, measures Measures, currentTime TimeFunc) (*TimeElapsedParser, error) {
```

Comment:
Once you have a few time elapsed parsers, I'm not sure that it will make sense which one is logging? It might be nice to store a logger for the TimeElapsedParser either by providing the `config.Name` to [`logger.Named()`](https://pkg.go.dev/go.uber.org/zap#Logger.Named) or providing it as a field using [`logger.With()`](https://pkg.go.dev/go.uber.org/zap#Logger.With).

Reply:
I'm doing that here in `provide.go`: https://github.com/xmidt-org/glaukos/blob/refactor/lib/eventmetrics/parsers/provide.go#L108=L109 but you're right it would be better to move that directly into the constructor function!